wlauto/common/android/workload.py

```diff
@@ -494,6 +515,22 @@ def reset(self, context):
     def run(self, context):
         ReventWorkload.run(self, context)
 
+        # state detection check if it's enabled in the config
```

Comment:
This whole section is basically a duplicate of the above, split it out into a function